internal/helm/helm.go

```diff
@@ -98,12 +100,32 @@ func Deploy(logger logr.Logger, parameters ChartParameters) error {
 
 	routesYaml := "~"
 	if len(parameters.Routes) > 0 {
+
+		logger.Info("routes and domains")
+
 		rs := []string{}
 		for _, desired := range parameters.Routes {
 			r := routes.FromString(desired)
-			rs = append(rs, fmt.Sprintf(`{"id":"%s","domain":"%s","path":"%s"}`,
-				strings.ReplaceAll(r.String(), "/", "."),
-				r.Domain, r.Path))
+			rdot := strings.ReplaceAll(r.String(), "/", ".")
+
+			domainSecret, err := domain.MatchDo(r.Domain, parameters.Domains)
+
+			logger.Info("domain match", "domain", r.Domain, "secret", domainSecret, "err", err)
+
+			// Should we treat a match error as something to stop for?
+			// The error can only come from `filepath.Match()`
+			var routeInfo string
+			if err != nil || domainSecret == "" {
+				// No secret found, no secret passed
+				routeInfo = fmt.Sprintf(`{"id":"%s","domain":"%s","path":"%s"}`,
+					rdot, r.Domain, r.Path)
+			} else {
+				// Pass the found secret
+				routeInfo = fmt.Sprintf(`{"id":"%s","domain":"%s","path":"%s","secret":"%s"}`,
+					rdot, r.Domain, r.Path, domainSecret)
+			}
```

Comment:
I'm not a big fan of handwritten JSON, but I see that we have an entire YAML a few lines after, so... let it be.
I will probably open a couple of issue/enhancement to do to move these to structs and/or use some libraries to do the marshaling.

Reply:
Yeah, this was keeping with the existing code (which IIRC I am also responsible for, i.e. wrote it :( ). Cleanup issues to come are good.